cc/crypto/encryptor.h

```diff
@@ -67,6 +73,38 @@ class ClientEncryptor {
   std::unique_ptr<SenderResponseContext> sender_response_context;
 };
 
+// Encryptor class for decrypting client requests that are received by the server and encrypting
+// server responses that will be sent back to the client. Each Encryptor corresponds to a single
+// crypto session between the client and the server.
+//
+// Sequence numbers for requests and responses are incremented separately, meaning that there could
+// be multiple responses per request and multiple requests per response.
+class ServerEncryptor {
+ public:
+  ServerEncryptor(KeyPair server_key_pair);
+
+  // Decrypts a [`EncryptedRequest`] proto message using AEAD.
+  // <https://datatracker.ietf.org/doc/html/rfc5116>
+  //
+  // `encrypted_request` must be a serialized [`oak.crypto.EncryptedRequest`] message.
+  // Returns a response message plaintext and associated data.
```

Comment:
The return value is not great, should there be a struct holding together both values? and use that for encryption too

Reply:
We won't be able to use the same struct for both result and input argument, because the input is a `absl::string_view` and the output is a `std::string`. We'll have to always copy the values into the new struct to send it to the encryptor as an argument.